dango/types/src/oracle/pyth.rs

```diff
@@ -0,0 +1,235 @@
+use {
+    super::{GuardianSetInfo, WormholeVAA},
+    anyhow::bail,
+    byteorder::BigEndian,
+    grug::{AddrEncoder, Api, Binary, BlockInfo, EncodedBytes, Inner, Map, Storage},
+    pyth_sdk::{Price, PriceFeed, PriceIdentifier},
+    pyth_wormhole::{BatchPriceAttestation, PriceStatus},
+    pythnet_sdk::{
+        accumulators::merkle::MerkleRoot,
+        hashers::keccak256_160::Keccak160,
+        messages::Message,
+        wire::{
+            from_slice,
+            v1::{AccumulatorUpdateData, Proof, WormholeMessage, WormholePayload},
+        },
+    },
+    serde::{de::Visitor, Deserialize},
+    std::str::FromStr,
+};
+
+pub const PYTHNET_ACCUMULATOR_UPDATE_MAGIC: &[u8; 4] = b"PNAU";
+
+macro_rules! uncast_enum {
+    ($data:expr, $name:path, $($params:ident),*) => {
+        match $data {
+            $name { $($params),* } => {
+                ($($params),*)
+            },
+        }
+    };
+    ($data:expr, $name:path) => {
+        match $data {
+            $name(t) => {
+                t
+            },
+        }
+    };
+}
+
+pub type PythId = EncodedBytes<[u8; 32], AddrEncoder>;
+
+#[derive(serde::Serialize, Clone, Debug, PartialEq, Eq)]
+pub struct PythVaa {
```

Comment:
we have an inconsistency in naming: there's `PythVaa` and there's `WormholeVAA`.

`Vaa` is more idiomatic than `VAA`.